index.html

```diff
@@ -5007,13 +4989,281 @@ <h3 id="looping-AudioBufferSourceNode">
+            <em>playback signal function</em> for <code>buffer</code>, which is
+            a function that maps from a <em>playhead position</em> to a set of
+            output signal values, one for each output channel. If
+            <code>position</code> corresponds to the location of an exact
+            sample frame in the buffer, this function returns that frame.
+            Otherwise, its return value is determined by a UA-supplied
+            algorithm that interpolates between sample frames in the
+            neighborhood of <code>position</code>. If <code>position</code> is
+            greater than or equal to <code>actualLoopEnd</code> and there is no
+            subsequent sample frame in <code>buffer</code>, then the next
+            sample frame should be taken as the playback signal at
+            <code>actualLoopStart</code> for interpolation purposes.
+            </li>
+            <li>Let the operation <em>optimize the buffer</em> be any operation
+            that alters both the buffer contents and sample rate in a way that
+            increases the efficiency or quality of rendering, while minimizing
+            changes to the <code>playbackSignal()</code> function. The nature
+            of this operation is up to the UA. Examples of such operations
+            might include upsampling, downsampling, applying a subsample
+            offset, or loop unrolling. This operation may affect the values of
+            <code>sampleRate</code>, <code>playbackRate</code> and
+            <code>offset</code>.
+            </li>
+            <li>Let <code>bufferTime</code> equal -1. This is the <em>playhead
+            position</em> within <code>buffer</code> of the next output sample
+            frame. The initial value -1 indicates that this position has not
+            yet been determined.
+            </li>
+            <li>
+              <em>Optimize the buffer</em> prior to rendering, if desired. For
+              example, if <code>playbackRate</code> is a constant value other
+              than 1, resampling up front for this rate may eliminate the need
+              for interpolation later.
+            </li>
+            <li>For each block of audio to be rendered:
+              <ol>
+                <li>Let <code>currentTime</code> be the current time of the <a>
+                  AudioContext</a>.
+                </li>
+                <li>Let <code>dt</code> be 1 / (_context sample rate_).
+                </li>
+                <li>Let <code>index</code> be 0.
+                </li>
+                <li>Let <code>computedPlaybackRate</code> be <code>playbackRate
+                * pow(2, detune / 1200) * (bufferRate / contextRate)</code> .
+                This combines the two k-rate parameters affecting playback rate
+                into a single computed quantity, and also corrects for any
+                difference in sample rate between the buffer and the
```

Comment:
As per the preceding comment, if the UA internally resamples the buffer, then I think both playbackRate and bufferRate should be adjusted so that computedPlaybackRate remains constant. That was why I suggested that optimization could affect these quantities.

If we don't include this ratio here, then how can we account for the fact that the effective playback rate must be unequal to 1 (and require interpolation) when the buffer rate and the sample rate are not identical?